test/fluree/db/query/misc_queries_test.clj

```diff
@@ -390,3 +391,174 @@
+  (testing "SCI evaluation of datatype and iri functions in filter expressions"
+    (let [conn   (test-utils/create-conn)
+          ledger @(fluree/create conn "sci-datatype-test")
+          ;; Insert some test data with string values
+          db     @(fluree/update (fluree/db ledger)
+                                 {"@context" {"ex" "http://example.org/"}
+                                  "insert"   [{"@id" "ex:existingOrNewData"
+                                               "ex:name" "Test Data"
+                                               "ex:description" "A test entity"}
+                                              {"@id" "ex:newData"
+                                               "ex:label" "New Data"
+                                               "ex:comment" "First new item"}
+                                              {"@id" "ex:newData2"
+                                               "ex:title" "Second Item"
+                                               "ex:note" "Another test"}
+                                              {"@id" "ex:newData3"
+                                               "ex:name" "Third Item"}
+                                              {"@id" "ex:newData5"
+                                               "ex:description" "Fifth Item"}
+                                              {"@id" "ex:newData6"
+                                               "ex:label" "Sixth Item"
+                                               "ex:comment" "Last item"}]})]
+
+      ;; Test the complex query with values, datatype bind, and filter
+      (let [query {"values" ["?s"
+                             [{"@type" "@id", "@value" "http://example.org/existingOrNewData"}
+                              {"@type" "@id", "@value" "http://example.org/newData"}
+                              {"@type" "@id", "@value" "http://example.org/newData2"}
+                              {"@type" "@id", "@value" "http://example.org/newData3"}
+                              {"@type" "@id", "@value" "http://example.org/newData5"}
+                              {"@type" "@id", "@value" "http://example.org/newData6"}]]
+                   "select" ["?s" "?property" "?value"]
+                   "groupBy" ["?s" "?property"]
+                   "where" [{"@id" "?s"
+                             "?property" "?value"}
+                            ["bind" "?dt" "(datatype ?value)"]
+                            ["filter" "(or (= ?dt \"http://www.w3.org/2001/XMLSchema#string\") (= ?property (iri \"@type\")))"]]
+                   "@context" {"ex" "http://example.org/"}}
+            result @(fluree/query db query)]
+
+        ;; Verify we got results
+        (is (pos? (count result))
+            "Should return results for string properties")
+
+        ;; All results should have string values or be @type predicates
+        ;; Note: groupBy wraps values in arrays
+        (is (every? #(let [value (nth % 2)]
```

Comment:
Instead of using `nth` here, can we destructure the whole tuple in the `let` binding for `value` and give proper names to the items we're checking?